cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs

```diff
@@ -232,3 +241,23 @@ instance ToObject (ChainIndepState (BlockProtocol blk))
       => ToObject (WrapChainIndepState blk) where
     toObject verb = toObject verb . unwrapChainIndepState
 
+instance FromJSON PortNumber where
+  parseJSON (Number portNum) = case readMaybe . show $ coefficient portNum of
+                                 Just port -> pure port
+                                 Nothing -> panic $ (Text.pack $ show portNum)
+                                                  <> " is not a valid port number."
+  parseJSON invalid  = panic $ "Parsing of port number failed due to type mismatch. "
+                             <> "Encountered: " <> (Text.pack $ Prelude.show invalid)
+
+-- Duplicated in cardano-cli
+
+instance ToJSON (HeaderHash blk) => ToJSON (Tip blk) where
+  toJSON TipGenesis = object [ "genesis" .= True ]
+  toJSON (Tip slotNo headerHash blockNo) =
+    object
+      [ "slotNo"     .= slotNo
+      , "headerHash" .= headerHash
+      , "blockNo"    .= blockNo
+      ]
+
+deriving newtype instance ToJSON BlockNo
```

Comment:
These ones are nothing to do with the hard fork infrastructure. All of these types are exported from the ouroboros-network package I think, so these instances should go in Cardano.TracingOrphanInstances.Network.